alvr/client_core/src/graphics/mod.rs

```diff
@@ -239,6 +304,46 @@ impl GraphicsContext {
             egl_context,
             _dummy_surface: dummy_surface,
             gl_context,
+            create_image,
+            destroy_image,
+            get_native_client_buffer,
+            image_target_texture_2d,
+        }
+    }
+
+    /// # Safety
+    /// `buffer` must be a valid AHardwareBuffer.
+    /// `texture` must be a valid GL texture.
+    pub unsafe fn render_ahardwarebuffer_using_texture(
+        &self,
+        buffer: *const c_void,
+        texture: gl::Texture,
+        render_cb: impl FnOnce(),
+    ) {
+        if !buffer.is_null() {
+            let client_buffer = (self.get_native_client_buffer)(buffer);
+            check_error(&self.gl_context, "get_native_client_buffer");
+
+            let image = (self.create_image)(
```

Comment:
I don't like that we're continually creating and destroying images

Reply:
This is what we always did. Apparently this is the pattern used both by other opengl and vulkan examples. Technically under the hood this is not too bad. Images are just a handle and the memory is just reused from the decoder android buffer.

Reply:
@barnabwhy there is no image to clear, the memory keeps being swapped

Reply:
It probably doesn't really matter for OpenGL especially, so sure